eventeditor/actor_json.py

```diff
@@ -0,0 +1,121 @@
+    Query = 1
+
+def load_actor_json(actor_name: str) -> typing.Dict[str, typing.Any]:
+    if not _actor_json_path:
+        return None
+
+    try:
+        # First look for individual actor file (overrides; for power users)
+        with open(_actor_json_path.parent/f'{actor_name}.json', 'rt') as file:
+            return json.loads(file.read())
+    except:
+        try:
+            # Otherwise look in actor definitions file
+            with open(_actor_json_path, 'rt') as file:
+                return json.loads(file.read())[actor_name]
+        except:
+            return None
+
+def load_event_parameters(actor_name: str, event_name: str, event_type: EventType) -> typing.Dict[str, typing.Any]:
+    try:
+        actor = load_actor_json(actor_name)
+
+        if event_type == EventType.Action:
+            return actor['actions'][event_name]
+        if event_type == EventType.Query:
+            return actor['queries'][event_name]
+        return None
+    except:
+        return None
+
+def load_actions(actor_name: str) -> typing.KeysView[str]:
+    try:
+        return load_actor_json(actor_name)['actions'].keys()
+    except:
+        return None
+
+def load_queries(actor_name: str) -> typing.KeysView[str]:
+    try:
+        return load_actor_json(actor_name)['queries'].keys()
+    except:
+        return None
+
+def export_definitions(flow, widget) -> None:
+    if not flow:
+        q.QMessageBox.information(widget, 'Export actor definition data', 'Open an event flow first')
+        return
+
+    if not _actor_json_path:
+        set_actor_json_path(q.QFileDialog.getSaveFileName(widget, 'Set ',  'actor_definitions', 'JSON (*.json)')[0])
+
+    if not _actor_json_path:
+        return
+
+    try:
+        with open(_actor_json_path, 'rt') as file:
+            definitions = json.loads(file.read())
+    except:
+        definitions = dict()
+    
+    for actor in flow.flowchart.actors:
+        if actor.identifier.name not in definitions:
+            definitions[actor.identifier.name] = {}
+
+        if 'actions' not in definitions[actor.identifier.name]:
+            definitions[actor.identifier.name]['actions'] = {}        
+        for action in actor.actions:
+            if action.v not in definitions[actor.identifier.name]['actions']:
+                definitions[actor.identifier.name]['actions'][action.v] = {}
+
+        if 'queries' not in definitions[actor.identifier.name]:
+            definitions[actor.identifier.name]['queries'] = {}
+        for query in actor.queries:
+            if query.v not in definitions[actor.identifier.name]['queries']:
+                definitions[actor.identifier.name]['queries'][query.v] = {}
+        
+        #! Should actor parameters also be supported?
+        #   - currently no auto-complete option at all?
+    
+    #! Should exporting be restructured to only loop through the events?
+    for event in flow.flowchart.events:
+        if hasattr(event.data, 'actor_action'):
```

Comment:
Check the event types instead of checking for the existence of specific property names (which is more fragile, doesn't really work with mypy and will break silently if we ever rename the actor_action field -- the worst kind of breakage).